### ISSUE TITLE
Closes point outside agent radius padding

### ISSUE THREAD

**Issue by claudijo** [vleue/vleue_navigator#85]
Is it possible to get the point closest to an obstacle outside the agent radius padding?

Use case is a point and click type of game where navigable spaces between obstacles are quite narrow due to padding added by the agent radius. 

The player might click on a point that looks to be between obstacles, but if that point is closer than the agent radius to an obstacle, navigation will be rejected, even if it is outside the actual obstacle. 

This behavior is kind of expected from a technical point of view. However, it would be nice in that case  to have the agent navigate as close as possible to the clicked point, not leaving the player frustrated because he/she thought they clicked on a valid space between obstacles.

**Comment by claudijo** [comment]
I think I found an ok solution.

Basically increasing setting values for `default_search_steps` and `default_search_delta`, and calling `get_closest_point` on the underlaying polyanya mesh, as in the example below.

```rust
pub fn compute_path(
    mut commands: Commands,
    target_query: Query<(Entity, &Target, &Transform), Changed<Target>>,
    meshes: Res<Assets<NavMesh>>,
    mesh_layout: Res<MeshLayout>,
) {
    for (entity, target, transform) in &target_query {
        let Some(mesh) = meshes.get(target.nav_mesh_id) else {
            continue;
        };

        let in_mesh =
            transform.translation.truncate() + mesh_layout.half_size() - mesh_layout.center();
        let to = target.target + mesh_layout.half_size() - mesh_layout.center();

        let Some(closest_point_in_mesh) = mesh.get().get_closest_point(to) else {
            continue;
        };

        let mesh = mesh.clone();
        let finding = FindingPath(Arc::new(RwLock::new((None, false))));
        let writer = finding.0.clone();
        AsyncComputeTaskPool::get()
            .spawn(async move {
                let path = mesh.path(in_mesh, closest_point_in_mesh.position());
                *writer.write().unwrap() = (path, true);
            })
            .detach();

        commands.entity(entity).insert(finding);
    }
}
```

Compare to the example: https://github.com/vleue/vleue_navigator/blob/e202d42d8825c5c2e8533cc9792f4d37c463b1e8/examples/moving.rs#L339

**Comment by mockersf** [comment]
> Basically increasing setting values for `default_search_steps` and `default_search_delta`

Yup, that's what they're for! It will search for the nearest point around circles of increasing radius by `default_search_delta`, with at most `default_search_steps` steps

You shouldn't have to call `get_closest_point`, searching for a path will do it. If you prefer to call it yourself, pass directly the resulting `closest_point_in_mesh` to the `path` function: it's a type `Coords` that already has some information about the point in the navmesh and will avoid redoing some searching. So `let path = mesh.path(in_mesh, closest_point_in_mesh);` should work and be a tiny bit faster.

**Comment by claudijo** [comment]
Thanks for the feedback. 

Alright, I see. You can use the polyanya::Coord value instead of a `Vec2` if you call `path` on the underlaying polyanya mesh as in `mesh.get().path(in_mesh, closest_point_in_mesh)`.

Not sure if I missed anything, but I noticed that if not calling `get_closest_point` and instead directly search for a path (after increasing values for `default_search_delta` and `default_search_steps`) does not quite give the same result, as it can generate a path with points that are closer to the obstacle than the agent radius. 

Only when I use `closest_point_in_mesh` I'm able to point give an arbitrary point inside the agent radius padding and get back a point that is as close as possible to the obstacle but not inside the agent radius padding.
